week2/homework/src/index.js

```diff
@@ -1,3 +1,125 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+const fs = require('fs');
+const fileName = './index.txt';
+const defaultEncoding = 'UTF8';
+
+const program = require('commander');
+
+program
+    .version('0.0.1');
+program
+    .command('list')
+    .description('Shows current to-dos')
+    .action(() => listItems());
+program
+    .command('add <index>')
+    .description('add a item to the list')
+    .action(index => addItem(index));
+program
+    .command('remove <index>')
+    .description('remove a item from list')
+    .action(index => removeItem(index));
+program
+    .command('reset')
+    .description('removes all the items from the list')
+    .action(() => resetList());
+program
+    .command('update <index> <newItem>')
+    .description('update a item from the list')
+    .action((index, updatedTask) => updateItem(index, updatedTask));
```

Comment:
This is and advanced topic and it's usually hard to understand at first. Don't worry if you don't get it or if it feels wrong. And feel free to ask questions if you have any :)

```js
action((index, updatedTask) => updateItem(index, updatedTask));
```
can be written as
```js
action(updateItem)
```

This is one of the beautiful things about javascript, you can pass functions as arguments to other functions.
In this case `action` will call the function for you, you can simplify this code by removing the intermediate function `.action(() => listItems())` ==> `.action(listItems)`